posthog/__init__.py

```diff
@@ -3,423 +3,427 @@
-        key=key,
-        distinct_id=distinct_id,
-        groups=groups,
-        person_properties=person_properties,
-        group_properties=group_properties,
-        only_evaluate_locally=only_evaluate_locally,
-        send_feature_flag_events=send_feature_flag_events,
-    )
-
-
-def get_all_flags(
-    distinct_id,  # type: str,
-    groups={},  # type: dict
-    person_properties={},  # type: dict
-    group_properties={},  # type: dict
-    only_evaluate_locally=False,  # type: bool
-):
-    """
-    Get all flags for a given user.
-    Example:
-    ```python
-    flags = posthog.get_all_flags('distinct_id')
-    ```
-
-    flags are key-value pairs where the key is the flag key and the value is the flag variant, or True, or False.
-    """
-    return _proxy(
-        "get_all_flags",
-        distinct_id=distinct_id,
-        groups=groups,
-        person_properties=person_properties,
-        group_properties=group_properties,
-        only_evaluate_locally=only_evaluate_locally,
-    )
-
-
-def get_feature_flag_payload(
-    key,
-    distinct_id,
-    match_value=None,
-    groups={},
-    person_properties={},
-    group_properties={},
-    only_evaluate_locally=False,
-    send_feature_flag_events=True,
-):
-    return _proxy(
-        "get_feature_flag_payload",
-        key=key,
-        distinct_id=distinct_id,
-        match_value=match_value,
-        groups=groups,
-        person_properties=person_properties,
-        group_properties=group_properties,
-        only_evaluate_locally=only_evaluate_locally,
-        send_feature_flag_events=send_feature_flag_events,
-    )
-
-
-def get_all_flags_and_payloads(
-    distinct_id,
-    groups={},
-    person_properties={},
-    group_properties={},
-    only_evaluate_locally=False,
-):
-    return _proxy(
-        "get_all_flags_and_payloads",
-        distinct_id=distinct_id,
-        groups=groups,
-        person_properties=person_properties,
-        group_properties=group_properties,
-        only_evaluate_locally=only_evaluate_locally,
-    )
-
-
-def page(*args, **kwargs):
-    """Send a page call."""
-    _proxy("page", *args, **kwargs)
-
-
-def screen(*args, **kwargs):
-    """Send a screen call."""
-    _proxy("screen", *args, **kwargs)
-
-
-def flush():
-    """Tell the client to flush."""
-    _proxy("flush")
-
-
-def join():
-    """Block program until the client clears the queue"""
-    _proxy("join")
-
-
-def shutdown():
-    """Flush all messages and cleanly shutdown the client"""
-    _proxy("flush")
-    _proxy("join")
-
-
-def _proxy(method, *args, **kwargs):
-    """Create an analytics client if one doesn't exist and send to it."""
-    global default_client
-    if disabled:
-        return None
-    if not default_client:
-        default_client = Client(
-            api_key,
+class Posthog(object):
+    log = logging.getLogger("posthog")
+
+    """Settings."""
+    disabled = False  # type: bool
+    _client: Client = None  # type: Client
```

Comment:
We don't need a Client proxy anymore, just inherit from Client

Reply:
Right now it's a bit annoying because we _still_ have to remember to add new functions here for it to work

Reply:
Fair point. I left it to start because it seems to be a good way to bound the functions we want exposed